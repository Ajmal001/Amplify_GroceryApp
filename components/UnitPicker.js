import React from 'react'
import { StyleSheet, Picker } from 'react-native'
import { grey } from '../utils/colors'
import PropTypes from 'prop-types'


export default function UnitPicker({ selectedValue, onValueChange, value, units}) {
    return (
         <Picker
        selectedValue={selectedValue}
        style={styles.picker}
        onValueChange={onValueChange}
        >
        <Picker.Item label={value} value={value} />
        {units.filter(unit => unit !== value).map(unit => (
        <Picker.Item label={unit} value={unit} key={unit} />
        ))}
        
    </Picker>
    )
}




const styles = StyleSheet.create({
    picker: {
        height: 40,
        borderColor: grey,
        borderWidth: 2,
        borderRadius: 10,
        paddingLeft: 15,
        marginTop: 10,
        marginLeft: 30,
        marginRight: 30,
    },
});

UnitPicker.propTypes = {
    selectedValue: PropTypes.string.isRequired,
    onValueChange: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired,
    units: PropTypes.array.isRequired,
}