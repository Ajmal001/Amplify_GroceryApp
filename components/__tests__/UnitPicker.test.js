import React from 'react';
import renderer from 'react-test-renderer';
import UnitPicker from '../UnitPicker';

test('renders correctly', () => {
  const tree = renderer.create(<UnitPicker
    selectedValue={'ct'}
    onValueChange={(x) => x}
    value={'kg'}
    units={['ct','kg']} />).toJSON();
  expect(tree).toMatchSnapshot();
});