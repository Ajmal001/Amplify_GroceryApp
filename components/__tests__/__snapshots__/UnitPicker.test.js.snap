// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#5c626b",
      "borderRadius": 10,
      "borderWidth": 2,
      "height": 40,
      "marginLeft": 30,
      "marginRight": 30,
      "marginTop": 10,
      "paddingLeft": 15,
    }
  }
>
  <RCTPicker
    items={
      Array [
        Object {
          "label": "kg",
          "textColor": undefined,
          "value": "kg",
        },
        Object {
          "label": "ct",
          "textColor": undefined,
          "value": "ct",
        },
      ]
    }
    onChange={[Function]}
    selectedIndex={1}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
